packages/babel-preset-expo/index.js

```diff
@@ -1,31 +1,150 @@
 module.exports = function(api) {
   const isWeb = api.caller(isTargetWeb);
 
-  const plugins = [
-    [
-      'babel-plugin-module-resolver',
-      {
-        alias: {
-          'react-native-vector-icons': '@expo/vector-icons',
-        },
-      },
-    ],
-    ['@babel/plugin-proposal-decorators', { legacy: true }],
-  ];
-
-  let disableImportExportTransform = undefined;
-  // On web add the `react-native-web` plugin.
   if (isWeb) {
-    plugins.push('babel-plugin-react-native-web');
-    // disableImportExportTransform = true;
+    const defaultPlugins = [
+      [
+        'babel-plugin-module-resolver',
+        {
+          alias: {
+            'react-native-vector-icons': '@expo/vector-icons',
+            /** Alias direct react-native imports to react-native-web */
+            'react-native$': 'react-native-web',
+            /** Add polyfills for modules that react-native-web doesn't support */
+            'react-native/Libraries/Image/AssetSourceResolver$':
+              'expo-asset/build/AssetSourceResolver',
+            'react-native/Libraries/Image/assetPathUtils$': 'expo-asset/build/Image/assetPathUtils',
+            'react-native/Libraries/Image/resolveAssetSource$':
+              'expo-asset/build/resolveAssetSource',
+          },
+        },
+      ],
+      ['@babel/plugin-syntax-flow'],
+      ['@babel/plugin-proposal-optional-catch-binding'],
```

Comment:
We shouldn't need most of these plugins since babel-preset-env has them (I sent you the list of them)